src/main/org/codehaus/groovy/ast/VariableScope.java

```diff
@@ -178,4 +178,26 @@ public Object removeReferencedClassVariable(String name) {
     public Iterator<Variable> getReferencedClassVariablesIterator() {
         return getReferencedClassVariables().values().iterator();
     }
-}
\ No newline at end of file
+
+    /**
+     * Gets a map containing the variables declared in this scope.
+     * This map cannot be modified.
+     * @return a map containing the declared variable references
+     */
+    public Map<String, Variable> getDeclaredVariables() {
```

Comment:
In javadoc, it says "This map cannot be modified" but this method can return two type of Map implementations: HashMap (see putDeclaredVariable method in the same file) and UnmodifiableMap stated in 'else' block.